rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/InfoView.java

```diff
@@ -267,27 +283,106 @@ private String niceUsageString(Map<String, String> usage) {
         }
     }
 
+    private LinkedHashMap<String, LinkedHashMap<String, String>> getAdditionalInformationAsMapOfMap(NodeSource ns) {
+        // We build a String of additional information to compare it with allAdditionalInformationAsString
+        // and update or not rm portal
+        String allNewAdditionalInformationAsString = "";
+
+        // Fill additionalInformationMap with new additional information (from ns)
+        LinkedHashMap<String, LinkedHashMap<String, String>> additionalInformationMap = new LinkedHashMap<>();
+
+        for (Map.Entry<String, String> categoryAndKeyAndValue : ns.getAdditionalInformation().entrySet()) {
+
+            // (key,value) in additionalInformation map: ("Azure Billing|Currency","USD")
+            // category: "Azure Billing"
+            // key: "Currency"
+            // value: "USD"
+            String[] categoryAndKeyArr = categoryAndKeyAndValue.getKey().split("\\|");
```

Comment:
you should call "continue" when categoryAndKeyArr  is not length = 2

Reply:
ok